pydrive2/files.py

```diff
@@ -97,6 +98,48 @@ def _GetList(self):
         return result
 
 
+class FakeWriteable(object):
```

Comment:
make it internal to the Readable

better name - it's not Fake - it's in-memory

is there something similar already in Python - some IOStringBuffer that you can just set to 0 when needed?

Reply:
`StringIO` and variants exist but they are much more heavyweight (and slower) than what's required; which is literally just saving the argument to `write()` for later use. It would be a pain to constantly `seek(0)` and `truncate()` a full buffer-like object. This object is only for internal use and isn't exposed in the API to an end-user. Perhaps it may even be an idea to define it in-line where it's actually used?

Reply:
Also it's fine if the user mutates the cached/buffer variable `chunk` as defined here since said object isn't ever internally mutated or read subsequently.

Reply:
I wonder though if we can save some time (avoid allocating buffer over and over again) if we use StringIO and use seek?

Reply:
ah ofc, the raw data bytestring will be deepcopied rather than saved by reference.

```python
class IoBuffer(object):
    """Lightweight retention of one chunk."""
    def write(self, chunk):
        self.chunk = chunk  # deepcopy

def _retry_request():
    return 200, b"some data"

fd = IoBuffer()
resp, chunk = _retry_request()
fd.write(chunk)
chunk += b" mutated"
print(fd.chunk)  # b"some data mutated"
```

`StringIO` is annoying though; we'd have to be careful about edge case (end of download) where a chunk might be smaller. We'd have to:

```python
fd = StringIO()
...
    fd.write(b"some data")
...
    fd.write(b"shorter")
    ...
    return fd.getvalue()[:fd.tell()]
```

Reply:
then again... `getvalue()` will return a deepcopy too...